include/swift/AST/Types.h

```diff
@@ -3547,6 +3550,20 @@ class SILFunctionType final : public TypeBase, public llvm::FoldingSetNode,
   /// Self type.
   ClassDecl *getWitnessMethodClass(ModuleDecl &M) const;
 
+  /// If this is a @convention(witness_method) function, return the conformance
+  /// for which the method is a witness.
+  ProtocolConformanceRef getWitnessMethodConformance() const {
+    assert(getRepresentation() == Representation::WitnessMethod);
+    return *WitnessMethodConformance;
+  }
+
+  /// If this is a @convention(witness_method) function, return the conformance
+  /// for which the method is a witness, if it isn't that convention, return
+  /// None.
+  Optional<ProtocolConformanceRef> getWitnessMethodConformanceOrNone() const {
```

Comment:
Is this variant useful?

Reply:
It's the one I call most often: it avoids having do a `fnTy->getRepresentation() == SILFunctionRepresentation::WitnessMethod ? fnTy.getWitnessMethodConformance() : None` dance.

Reply:
I see, OK.